chat/agent.js

```diff
@@ -1833,6 +1869,41 @@ window.menu = null;
                     console.warn('unexpected response', nn);
                   }
                 }, this.handleInput = function(n) {
+
+                  if(n && IS_STREAMING_ACTIVE) {
+                    // case: during a streaming user enter new message and press stop streaming button;
+                    var ty = "cancel streaming text"; 
+                    window.isChatting = true; 
+                    t.removeFeedbackWidget();
+                    IS_STREAMING_ACTIVE = false;
+                    window.eySocket.streamedMessages = [];
+                    window.eySocket.queuedMessages = [];
+                    window.eySocket.send(JSON.stringify(t.buildPayLoad("", ty)));
+                    window.isChatting = false; 
+                    window.parent.postMessage(ty, "*"); 
+                    t.removeStopStreamingButton()
+                    t.domHelper.handleStartSend()
+                    return;
+                  }
+
+                  if (!n && IS_STREAMING_ACTIVE) {
+                    // case: stop streaming and message is empty;
+                    var ty = "cancel streaming text"; 
+                    window.isChatting = true; 
+                    t.removeFeedbackWidget();
+                    IS_STREAMING_ACTIVE = false;
+                    window.eySocket.streamedMessages = [];
+                    window.eySocket.queuedMessages = [];
+                    window.eySocket.send(JSON.stringify(t.buildPayLoad("", ty)));
+                    t.domHelper.setInputValue(""); 
```

Comment:
you could probably combine these with if (IS_STREAMING_ACTIVE)

then use if (!n) inside the if statement.

Reply:
fixed